content/en/blog/_posts/XX-userns-enabled-by-default/index.md

```diff
@@ -0,0 +1,280 @@
+
+The file-systems that need to support idmap mounts are all the file-systems used
+by a pod in the `pod.spec.volumes` field.
+
+This means: for PV/PVC volumes, the file-system used in the PV needs to support
+idmap mounts; for hostPath volumes, the file-system used in the hostPath
+needs to support idmap mounts.
+
+What does this mean for secrets/configmaps/projected/downwardAPI volumes? For
+these volumes, the kubelet creates a `tmpfs` file-system. So, you will need a
+6.3 kernel to use these volumess (note that if you use them as env variables it
+is fine).
+
+And what about emptyDir volumes? Those volumes are created by the Kubelet by
+default in `/var/lib/kubelet/pods/`. You can also use a custom directory for
+this. But what needs to support idmap mounts is the file-system used in that
+directory.
+
+The kubelet creates some more files for the container, like `/etc/hostname`,
+`/etc/resolv.conf`, `/dev/termination-log`, `/etc/hosts`, etc. These files are
+also created in `/var/lib/kubelet/pods/` by default, so it's important for the
+file-system used in that directory to support idmap mounts.
+
+Also, some container runtimes may put some of these ephemeral volumes inside a
+`tmpfs` file-system, in which case you will need support for idmap mounts in
+`tmpfs`.
+
+**4. Can I use a kernel older than 6.3?**
+
+Yes, but you will need to make sure you are not using a `tmpfs` file-system. If
+you avoid that, you can easily use 5.19 (if all the other file-systems you use
+support idmap mounts in that kernel).
+
+To avoid using `tmpfs` can be tricky, though, as we just mentioned. But besides
+having to avoid those volume types, you will also have to avoid mounting the
+service account token. Every pod have it mounted by default, and it is using a
+projected volume that, as we mentioned, uses a `tmpfs` file-system.
+
+You can even go lower than 5.19, all the way to 5.12. However, your container
+rootfs is probably using an overlayfs file-system, and support for overlayfs was
+added in 5.19. We wouldn't recommend to use a kernel older than 5.19, as not
+being able to use idmap mounts for the rootfs is a big limitation. If you
+absolutely need to, you can check [this blog post][userns-tricks] Rodrigo wrote
+some years ago, about tricks to use user-namespaces when you we can't support
+idmap mounts on the rootfs.
+
+[userns-tricks]: https://kinvolk.io/blog/2023/11/tips-and-tricks-for-user-namespaces-with-kubernetes-and-containerd
+
+**5. If my stack supports user namespaces, do I need to configure anything else?**
+
+No, if your stack supports it and you are using Kubernetes 1.33, there is
+nothing you need to configure. You should be able to follow the task: [Use a
+user namespace with a pod][task-userns].
+
+However, there are some configuration you can do, you can check them
+[here][userns-k8s-conf]. You can also enable a [feature gate to relax the PSS
+rules][userns-pss].
```

Comment:
I found this part quite confusing -- you start off saying there is nothing needed. But then you tell me otherwise right after. I think it may be more friendly to start with something like "No, as long as you do not have peculiar requirements, you just need to use Kubernetes v1.33 to use user namespaces" -- and then follow up with some sample cases where configuration could be helpful?

Reply:
What is confusing exactly of that? Nothing is _needed_ for it to work, however there are configurations you _can_ change, but it's not required to set any configuration before using userns.

How would you say that?

Reply:
It's not that the content is confusing, but it sounds like you are contradicting yourself (you are surely not, but just in terms of the flow).

Maybe phrase it with something like "No, there is nothing needed to be configured ...
In case you have specific requirements, you may configure with various options. You can find more in..." ?

Reply:
Perfect, that is also better!

Thanks for all the comments, not just this, it read better now :)

Applied that and pushed now!